app/scripts/modules/core/cluster/onDemand/onDemandClusterPicker.component.ts

```diff
@@ -0,0 +1,62 @@
+import {IComponentController, IComponentOptions, IScope, module} from 'angular';
+import {Application} from '../../application/application.model';
+import {IClusterSummary} from '../../domain/ICluster';
+
+require ('./onDemandClusterPicker.component.less');
+
+class OnDemandClusterPickerController implements IComponentController {
+  public application: Application;
+  public availableClusters: IClusterSummary[];
+  public lastSelection: string;
+  public totalClusterCount: number;
+  public optionTemplate: string = require('./onDemandOptionTemplate.html');
```

Comment:
Since you already inlined the bigger template for the component, let's inline this one too.